encodings/alp/src/compute.rs

```diff
@@ -67,3 +80,84 @@ impl SliceFn for ALPArray {
         .into_array())
     }
 }
+
+impl FilterFn for ALPArray {
+    fn filter(&self, predicate: &Array) -> VortexResult<Array> {
+        Ok(Self::try_new(
+            filter(&self.encoded(), predicate)?,
+            self.exponents(),
+            self.patches().map(|p| filter(&p, predicate)).transpose()?,
+        )?
+        .into_array())
+    }
+}
+
+impl MaybeCompareFn for ALPArray {
+    fn maybe_compare(&self, array: &Array, operator: Operator) -> Option<VortexResult<Array>> {
+        if ConstantArray::try_from(array).is_ok()
+            || array
+                .statistics()
+                .get_as::<bool>(Stat::IsConstant)
+                .unwrap_or_default()
+        {
+            let rhs = scalar_at(array, 0).vortex_expect("should be scalar");
+            let pvalue = rhs
+                .value()
+                .as_pvalue()
+                .vortex_expect("Expected primitive value");
+
+            match pvalue {
+                Some(PValue::F32(f)) => {
+                    let encoded = f32::encode_single(f, self.exponents());
+                    match encoded {
+                        Ok(encoded) => {
+                            let s = ConstantArray::new(encoded, self.len());
+                            Some(compare(&self.encoded(), s.as_array_ref(), operator))
+                        }
+                        Err(exception) => {
+                            if let Some(patches) = self.patches().as_ref() {
+                                let s = ConstantArray::new(exception, self.len());
+                                Some(compare(patches, s.as_array_ref(), operator))
+                            } else {
+                                Some(Ok(BoolArray::from_vec(
+                                    vec![false; self.len()],
+                                    Validity::AllValid,
+                                )
+                                .into_array()))
+                            }
+                        }
+                    }
```

Comment:
I think if you make a function generic over ALPFloat you can reduce the repetition